logdetective/logdetective.py

```diff
@@ -2,248 +2,31 @@
-        for chunk in get_chunks(log):
-            procesed_line = self.miner.add_log_message(chunk)
-            LOG.debug(procesed_line)
-        sorted_clusters = sorted(self.miner.drain.clusters, key=lambda it: it.size, reverse=True)
-        for chunk in get_chunks(log):
-            cluster = self.miner.match(chunk, "always")
-            if cluster in sorted_clusters:
-                out += f"{chunk}\n"
-                sorted_clusters.remove(cluster)
-        return out
-
-
-def download_model(url: str, verbose: bool = False) -> str:
-    """ Downloads a language model from a given URL and saves it to the cache directory.
-
-    Args:
-        url (str): The URL of the language model to be downloaded.
-
-    Returns:
-        str: The local file path of the downloaded language model.
-    """
-    path = os.path.join(
-        os.path.expanduser(CACHE_LOC), url.split('/')[-1])
-
-    LOG.info("Downloading model from %s to %s", url, path)
-    if not os.path.exists(path):
-        if verbose:
-            path, _status = urlretrieve(url, path, MyProgressBar())
-        else:
-            path, _status = urlretrieve(url, path)
-
-    return path
-
-
-def process_log(log: str, model: Llama) -> str:
-    """
-    Processes a given log using the provided language model and returns its summary.
-
-    Args:
-        log (str): The input log to be processed.
-        model (Llama): The language model used for processing the log.
-
-    Returns:
-        str: The summary of the given log generated by the language model.
-    """
-    return model(PROMPT_TEMPLATE.format(log), max_tokens=0)["choices"][0]["text"]
-
-
-def retrieve_log_content(log_path):
-    """Get content of the file on the log_path path."""
-    parsed_url = urlparse(log_path)
-    log = ""
-
-    if not parsed_url.scheme:
-        if not os.path.exists(log_path):
-            raise ValueError(f"Local log {log_path} doesn't exist!")
-
-        with open(log_path, "rt") as f:
-            log = f.read()
-
-    else:
-        log = requests.get(log_path, timeout=60).text
-
-    return log
-
-
 def main():
     """Main execution function."""
     parser = argparse.ArgumentParser("logdetective")
-    parser.add_argument("url", type=str, default="")
-    parser.add_argument("-M", "--model", type=str, default=DEFAULT_ADVISOR)
-    parser.add_argument("-S", "--summarizer", type=str, default="drain")
-    parser.add_argument("-N", "--n_lines", type=int, default=5)
+    parser.add_argument("url", type=str, default="", help="The URL of the log file to be analyzed.")
+    parser.add_argument("-M", "--model", help="The path or URL of the language model for analysis.",
+                        type=str, default=DEFAULT_ADVISOR)
+    parser.add_argument("-S", "--summarizer", type=str, default="drain",
+                        help="Choose between LLM and Drain template miner as the log summarizer.\
+                                LLM must be specified as path to a model, URL or local.")
```

Comment:
I wonder what `local` means here.

Reply:
in the file system.

Reply:
I think it needs more explanation. Does it mean path to local model or name of the model in cache directory? If the second how it should be specified.

Could you please improve the help a bit?